src/networkd.c

```diff
@@ -967,6 +967,7 @@ write_networkd_conf(const NetplanNetDefinition* def, const char* rootdir)
 
     if (def->type == NETPLAN_DEF_TYPE_WIFI || def->has_auth) {
         g_autofree char* link = g_strjoin(NULL, rootdir ?: "", "/run/systemd/system/systemd-networkd.service.wants/netplan-wpa-", def->id, ".service", NULL);
+        g_autofree char* link_target = g_strjoin(NULL, rootdir ?: "", "/run/systemd/system/netplan.target.wants/netplan-wpa-", def->id, ".service", NULL);
```

Comment:
Since this facility is available, we should possibly also add `netplan.target.wants/systemd-networked.service` and `netplan.target.wants/systemd-wait-online.server`. This would enable us to have neither networkd nor NM enabled in the image, and cloud-init able to activate one or the other on boot.

Reply:
That should work. We would need some extra code to activate NetworkManager, as we're not setting up systemd units/dependencies from our NM renderer currently, but its certainly possible.

This would only work for cloud-init based images, though. As in a "normal" boot sequence `netplan.target` would never be called/activated.